packages/yourloops/components/dialogs/join-team/team-code-confirm.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { useTranslation } from 'react-i18next'
+import React, { createRef, useEffect, useState } from 'react'
+import { getDisplayTeamCode, REGEX_TEAM_CODE_DISPLAY, type Team, useTeam } from '../../../lib/team'
+import Box from '@mui/material/Box'
+import DialogTitle from '@mui/material/DialogTitle'
+import DialogContent from '@mui/material/DialogContent'
+import DialogContentText from '@mui/material/DialogContentText'
+import TextField from '@mui/material/TextField'
+import DialogActions from '@mui/material/DialogActions'
+import Button from '@mui/material/Button'
+import { useAlert } from '../../utils/snackbar'
+import ProgressIconButtonWrapper from '../../buttons/progress-icon-button-wrapper'
+
+export interface ConfirmTeamProps {
+  onClickCancel: () => void
+  onCompleteStep: (team: Team) => void
+  teamName: string
+}
+
+export const TeamCodeConfirm = (props: ConfirmTeamProps): JSX.Element => {
+  const { onClickCancel, onCompleteStep, teamName } = props
+  const { t } = useTranslation()
+  const inputRef = createRef<HTMLInputElement>()
+  const teamHook = useTeam()
+  const alert = useAlert()
+  const [numericCode, setNumericCode] = useState<string>('')
+  const [idCode, setIdCode] = useState<string>('')
+  const [isInProgress, setIsInProgress] = useState<boolean>(false)
+  const joinButtonDisabled = !idCode.match(REGEX_TEAM_CODE_DISPLAY)
+
+  const getNumericCode = (value: string): string => {
+    const code = []
+    const arrayString = value.split('')
+    arrayString.forEach((element) => {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏I suppose that this is old code, there are better ways of writing this.

Reply:
this is new because the old one was done with a for loop